### DISCUSSION TITLE
Runpod Users Here Using Stable Diffusion ?

### DISCUSSION THREAD

**Discussion by HelpANewbie** [TheLastBen/fast-stable-diffusion#2288] [General]
I wanted to know how fast do images get generated when using Runpod. - the 0.2 $ an hour one 

I've used the free versions on collab and paperspace - planning to shift to the paid version now hence wanted to know the ease of usage. (I dont understand codes etc - just basic click, type prompts etc) 

Looking forward

**Comment by TheLastBen** [comment]
around 10it/s with the 3070 (0.2$), 15it/s with the 3090 (0.29$)

**Reply by HelpANewbie** [reply]
10 it/s ? You mean 10 images per second ?

**Reply by HelpANewbie** [reply]
Also Ben...Is it as simple as Paperspace and Collab to run SD webui on run pod ?

**Reply by TheLastBen** [reply]
yes, the runpod notebooks are the same as paperspace

with colab T4 GPU, it's 6it/s

**Reply by HelpANewbie** [reply]
Thank you Ben...Do you know about Deforum...and if we can run that through Runpod too ?

**Reply by TheLastBen** [reply]
yes, simply install the extension

**Comment by HelpANewbie** [comment]
How do we add a model in the webui ? On runpod directly the webui opens when we click on connect and then http

**Comment by HelpANewbie** [comment]
in paperspace we can directly paste the google drive link of the model and then it loads so quickly....In Runpod how can I do that ? plz help

**Reply by TheLastBen** [reply]
paperspace notebook are exactly the same as Runpod notebooks, make sure you're using the template "fast-stable-diffusion" in Runpod https://www.runpod.io/console/gpu-cloud?template=runpod-stable-unified

**Comment by HelpANewbie** [comment]
yes its same...once we deploy the pod then we have to click on connect and then that http link which opens the webui right ?

**Comment by HelpANewbie** [comment]
Oh ok I need to open the jupiter notebook link and then run it through there....got it...thanks Ben...Ill try running it like that now :)

**Comment by HelpANewbie** [comment]
There is one issue....Im uploading the safetensors file...in paperspace we had to change one part from False to True....Here that option is not there...Its not loading the safetensors model file....how to do it ?

**Reply by TheLastBen** [reply]
it detects the file automatically

**Reply by HelpANewbie** [reply]
it gave me an error with the safetensors file - said it failed to load it ...i now uploaded another ckpt file so it is working with that but not with safetensors

**Reply by TheLastBen** [reply]
do you have the link ?

**Reply by TheLastBen** [reply]
what is the error message ?

**Reply by TheLastBen** [reply]
it's downloading without an issue, try removing the spaces in the filename

**Reply by HelpANewbie** [reply]
ill have to load the safetensors file again as i didnt save the error message...ill do it again now and past the error message here

**Reply by HelpANewbie** [reply]
oh ok let me try again

**Reply by TheLastBen** [reply]
the problem is the spaces in the filename

**Reply by TheLastBen** [reply]
rename it and it will work fine

**Reply by HelpANewbie** [reply]
ok ill try now

**Reply by HelpANewbie** [reply]
If I use a very high power GPU - does it affect the quality of the images ? I mean will the model make better images or only the time gets affected ?

**Reply by HelpANewbie** [reply]
I'm still getting this error : 

sage: webui.py [-h] [--update-all-extensions] [--skip-python-version-check]
                [--skip-torch-cuda-test] [--reinstall-xformers]
                [--reinstall-torch] [--update-check] [--test-server]
                [--skip-prepare-environment] [--skip-install]
                [--data-dir DATA_DIR] [--config CONFIG] [--ckpt CKPT]
                [--ckpt-dir CKPT_DIR] [--vae-dir VAE_DIR]
                [--gfpgan-dir GFPGAN_DIR] [--gfpgan-model GFPGAN_MODEL]
                [--no-half] [--no-half-vae] [--no-progressbar-hiding]
                [--max-batch-count MAX_BATCH_COUNT]
                [--embeddings-dir EMBEDDINGS_DIR]
                [--textual-inversion-templates-dir TEXTUAL_INVERSION_TEMPLATES_DIR]
                [--hypernetwork-dir HYPERNETWORK_DIR]
                [--localizations-dir LOCALIZATIONS_DIR] [--allow-code]
                [--medvram] [--lowvram] [--lowram]
                [--always-batch-cond-uncond] [--unload-gfpgan]
                [--precision {full,autocast}] [--upcast-sampling] [--share]
                [--ngrok NGROK] [--ngrok-region NGROK_REGION]
                [--ngrok-options NGROK_OPTIONS]
                [--enable-insecure-extension-access]
                [--codeformer-models-path CODEFORMER_MODELS_PATH]
                [--gfpgan-models-path GFPGAN_MODELS_PATH]
                [--esrgan-models-path ESRGAN_MODELS_PATH]
                [--bsrgan-models-path BSRGAN_MODELS_PATH]
                [--realesrgan-models-path REALESRGAN_MODELS_PATH]
                [--clip-models-path CLIP_MODELS_PATH] [--xformers]
                [--force-enable-xformers] [--xformers-flash-attention]
                [--deepdanbooru] [--opt-split-attention]
                [--opt-sub-quad-attention]
                [--sub-quad-q-chunk-size SUB_QUAD_Q_CHUNK_SIZE]
                [--sub-quad-kv-chunk-size SUB_QUAD_KV_CHUNK_SIZE]
                [--sub-quad-chunk-threshold SUB_QUAD_CHUNK_THRESHOLD]
                [--opt-split-attention-invokeai] [--opt-split-attention-v1]
                [--opt-sdp-attention] [--opt-sdp-no-mem-attention]
                [--disable-opt-split-attention] [--disable-nan-check]
                [--use-cpu USE_CPU [USE_CPU ...]] [--listen] [--port PORT]
                [--show-negative-prompt] [--ui-config-file UI_CONFIG_FILE]
                [--hide-ui-dir-config] [--freeze-settings]
                [--ui-settings-file UI_SETTINGS_FILE] [--gradio-debug]
                [--gradio-auth GRADIO_AUTH]
                [--gradio-auth-path GRADIO_AUTH_PATH]
                [--gradio-img2img-tool GRADIO_IMG2IMG_TOOL]
                [--gradio-inpaint-tool GRADIO_INPAINT_TOOL]
                [--gradio-allowed-path GRADIO_ALLOWED_PATH]
                [--opt-channelslast] [--styles-file STYLES_FILE]
                [--autolaunch] [--theme THEME] [--use-textbox-seed]
                [--disable-console-progressbars] [--enable-console-prompts]
                [--vae-path VAE_PATH] [--disable-safe-unpickle] [--api]
                [--api-auth API_AUTH] [--api-log] [--nowebui]
                [--ui-debug-mode] [--device-id DEVICE_ID] [--administrator]
                [--cors-allow-origins CORS_ALLOW_ORIGINS]
                [--cors-allow-origins-regex CORS_ALLOW_ORIGINS_REGEX]
                [--tls-keyfile TLS_KEYFILE] [--tls-certfile TLS_CERTFILE]
                [--disable-tls-verify] [--server-name SERVER_NAME]
                [--gradio-queue] [--no-gradio-queue] [--skip-version-check]
                [--no-hashing] [--no-download-sd-model] [--subpath SUBPATH]
                [--add-stop-route] [--controlnet-dir CONTROLNET_DIR]
                [--controlnet-annotator-models-path CONTROLNET_ANNOTATOR_MODELS_PATH]
                [--no-half-controlnet]
                [--controlnet-preprocessor-cache-size CONTROLNET_PREPROCESSOR_CACHE_SIZE]
                [--controlnet-loglevel {DEBUG,INFO,WARNING,ERROR,CRITICAL}]
                [--ldsr-models-path LDSR_MODELS_PATH] [--lora-dir LORA_DIR]
                [--scunet-models-path SCUNET_MODELS_PATH]
                [--swinir-models-path SWINIR_MODELS_PATH]
webui.py: error: unrecognized arguments: NEW.safetensors

**Reply by TheLastBen** [reply]
the GPU doesn't affect the quality, 
what is the name of the model ?

**Reply by TheLastBen** [reply]
rename it to `rfktrs_darkdream.safetensors`, no spaces in the filename

**Reply by HelpANewbie** [reply]
Oh ! Ok ill try now

**Reply by HelpANewbie** [reply]
Thankkkkkssss so much....Its working now :)

**Comment by HelpANewbie** [comment]
When using deforum - the pod shows as gpu not being used....its at 0% used while running/generation process - also its quite slow to generate the video....Wanted to know whats the best gpu to select for deforum.....if i select a A30 2x - does it make things generate faster ? is it better to use two gpus than 1 which might be of a higher capacity ? 

Sorry for many queries...If there is any link which answers all these things please do share...or if anyone can help sharing their experiences

**Reply by TheLastBen** [reply]
select the 3090 (0.20$/h), it's the best GPU

**Reply by HelpANewbie** [reply]
Thanks Ben...But if I want to increase the generation speed...will making it 2x work ?

**Reply by TheLastBen** [reply]
nope, it only uses one single GPU, so choosing an x2 GPU machine won't help

**Reply by HelpANewbie** [reply]
Oh alright...even the A100 or H100 won't make generations superfast ?

**Reply by TheLastBen** [reply]
A100 is slightly better but not by much, the H100 should be faster but it's 10 times more expensive

**Reply by HelpANewbie** [reply]
Yeah...so expensive...if it doesn't make much of a difference why are their prices so high ? Any idea ?

**Reply by TheLastBen** [reply]
They are designed to handler big tasks, using them for SD is a waste of money

**Comment by HelpANewbie** [comment]
@TheLastBen Do you know the difference between secure and community cloud on Runpod ? Which one's better ....any difference with respect to creation speeds ?

**Reply by TheLastBen** [reply]
go with community same speed and cheaper

**Comment by HelpANewbie** [comment]
@TheLastBen Im having one problem...there is a textual inversion file which is based on 2.0 768 stable diffusion - in this current one there is no option of that model of SD - i tried selecting the 2.1 768 and then uploaded the textual inversion file but it is not loading and saying that it is skipped....can you help please on this ?

**Reply by TheLastBen** [reply]
The textual inversion notebook is only for sd1.5

**Reply by HelpANewbie** [reply]
im not using that textual inversion notebook - i just load the textual inversion file in the embeddings in the first automatic 1111 notebook and then load the models but when it loads it shows as the file has skipped

**Reply by TheLastBen** [reply]
check with the A1111 repo since the issue isn't related to the notebook

**Reply by HelpANewbie** [reply]
i tried everything but its not working sadly
